client/verta/verta/registry/_verta_model_base.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+
+import abc
+
+from verta.external import six
+
+
+@six.add_metaclass(abc.ABCMeta)
+class VertaModelBase(object):
+    """Abstract base class for Verta Standard Models.
+
+    .. note::
+
+        ``__init__()`` and :meth:`predict` **must** be implemented by
+        subclasses. Any other methods are optional.
+
+    Parameters
+    ----------
+    artifacts : dict of str to str
+        A mapping of artifact keys to filepaths. This will be provided to the
+        deployed model based on artifact keys specified through
+        :meth:`RegisteredModelVersion.log_model()
+        <verta.registry.entities.RegisteredModelVersion.log_model>`.
+
+    Examples
+    --------
+    .. code-block:: python
+
+        import pickle
+        import numpy as np
+        from verta.registry import VertaModelBase
+
+        class Model(VertaModelBase):
+            def __init__(self, artifacts):
+                with open(artifacts["np_matrix"], "rb") as f:
+                    self._transform = pickle.load(f)
+
+            def predict(self, input):
+                input = np.array(input)
+
+                return np.matmul(input, self._transform)
+
+        # iterate locally
+        model = Model(
+            artifacts=model_ver.fetch_artifacts(["np_matrix"]),
+        )
+
+        # persist to model version
+        model_ver.log_model(Model, artifacts=["np_matrix"])
+
+    """
+
+    @abc.abstractmethod
```

Comment:
This doesn't seem to show on the docs  expected?

Reply:
Sphinx doesn't render `__init__()` by default. I can change our templates to do so, but it'll take time (maybe an hour?) if I need to change other docstrings to accommodate it.

How important would it be to show this?

Reply:
After a discussion, I have filed VR-11770 to investigate whether it would make sense to document __init__() separately; for now, it seems like the convention is to not.